staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -110,10 +99,48 @@ func main() {
 	signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-ch
-		log.Printf("Received termination, signaling shutdown")
+		klog.Info("Received termination, signaling shutdown")
 		cancel()
 	}()
 
+	if !leaderElect {
+		klog.Infof("Run without leader election")
+
+		go func(ctx context.Context) {
+			select {
+			case <-ctx.Done():
+				klog.Fatalf("Received termination, quiting")
+			}
+		}(ctx)
+
+		run(ctx)
+
+		panic("unreachable")
+	}
+
+	klog.Infof("Run with leader election")
+
+	id, err := os.Hostname()
```

Comment:
We always expect the hostname to be the same in this example since it'll be run from the same machine.